### PR TITLE
Dash api

### LINKED ISSUE #154: Implement an API to enable querying the S3 objects created by the data-pipelines
See: https://github.com/KlimaDAO/data-pipelines/edit/staging/PROJECT.md

### LINKED ISSUE #157: Dash api improvement: digital carbon pricing low expiry and 7 day change

### LINKED ISSUE #159: Dash api improvement: retirements: add number of transactions when aggregating

### LINKED ISSUE #160: Dash api improvement: retirements: add an aggregated view of carbon metrics accross chains

### LINKED ISSUE #161: Dash api improvement: retirements: auto select bridge when pool is selected

### LINKED ISSUE #162: Dash api improvement: retirements: autoselect date_field for daily aggregations

### PR DESCRIPTION
**Pull request by biwano**
Dash-API application

Fixes #154 
Fixes #157
Fixes #159 
Fixes #160
Fixes #161 
Fixes #162 

- [x] Raw Credits endpoint
- [x] Credits aggregations endpoints
  - [x] Total
  - [x] Daily aggregation (normal and cumulative)
  - [x] Monthly aggregation (normal and cumulative)
  - [x] Countries aggregation
  - [x] Projects aggregation
  - [x] Methodologies aggregation
 - [x] Raw pools endpoint
 - [x] Pools aggregations endpoint
  - [x] Total
  - [x] Daily aggregation (normal and cumulative)
  - [x] Monthly aggregation (normal and cumulative)
- [x] Credits Holders endpoint
- [x] Token prices endpoints
  - [x] Latest Price
  - [x] History
- [x] Carbon Supply endpoint
- [x] Raw Retirements endpoint
- [x] Retirements aggregations endpoints
  - [x] Pools aggregations
  - [x] Tokens aggregations
  - [x] Chains aggregations
  - [x] Beneficiaries aggregations

### CODE REVIEW

File: src/apps/services/cache.py [hunk 1]
```diff
@@ -0,0 +1,281 @@
...
+
+    def date_manipulations_daily(self, df, date_column):
+        if not (df.empty):
+            df[date_column] = (
+                df[date_column]
+                .dt.tz_localize(None)
+                .dt.floor("D")
+                .dt.date
+            )
+            """
```
Comment (0xAurelius) [inline src/apps/services/cache.py:253]: What's the deal with this docstring comment? Unnecessary right now? More functionality to be added later?

Should be a `TODO:` on it if so...

**Review by 0xAurelius — CHANGES_REQUESTED** [review]
Raised one minor issue with a stray docstring commented chunk of code

It does seem like we could rename the API endpoints to refer to `credits` instead of `offsets` and then later on deal with the change in the underlying subgraphs... that way when we build the new frontend that queries the API, it will already be referring to `credits` instead of `offsets` (even though the underlying datasets are still called `offsets`)

AFAICT none of the actual response contents use the term `offset` - only the endpoint names...

File: src/apps/services/cache.py (hunk 1 above)
Comment (biwano) [reply src/apps/services/cache.py:253]: Ho no my bad. I just forgot to remove it.

**Comment by biwano** [comment]
> Raised one minor issue with a stray docstring commented chunk of code
> 
> It does seem like we could rename the API endpoints to refer to `credits` instead of `offsets` and then later on deal with the change in the underlying subgraphs... that way when we build the new frontend that queries the API, it will already be referring to `credits` instead of `offsets` (even though the underlying datasets are still called `offsets`)
> 
> AFAICT none of the actual response contents use the term `offset` - only the endpoint names...

I changed the endpoints and refactored the code

**Review by 0xAurelius — APPROVED** [review]

**Pull request merged**